Interaction Events v1 (WIP)

i509VCB opened the pull request:
This is not complete yet, but available for viewing so people can comment on the new module as it comes together.

So far I have implemented the block break/place events for the logical server and other parts will be implemented over the next few days.

There are also other things being considered such as an Interact/Attack air event and other things discussed in #1247

This will resolve #1247

maityyy commented on fabric-interaction-events-v1/src/main/java/net/fabricmc/fabric/impl/interaction/InteractionRaycasting.java:
```diff
@@ -0,0 +1,109 @@
+	 * @param player the player
+	 * @return the reach distance
+	 */
+	public static double getReachDistance(ServerPlayerEntity player) {
```
Why `ServerPlayerEntity` and not just `PlayerEntity`?

i509VCB replied:
I don't believe that will be needed on the logical client, but time will tell there

TheGlitch76 commented on fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/api/event/client/player/ClientPickBlockCallback.java:
```diff
@@ -20,14 +20,16 @@
 /**
  * This event handler has been deprecated due to not hooking nicely
- * into the game. Please use the alternatives.
+ * into the game and is no longer implemented. Please use the alternatives provided in {@link ClientPickItemEvents}.
```
If this is no longer implemented, it needs to be removed, full stop.

Silent failing bugs are *impossible* to track down, please do not introduce one into fabric api.

OroArmor commented:
I think some of the combat events should be added to its own module: `fabric-combat-api-v1`. I have a start here: https://github.com/FabricMC/fabric/pull/1210

i509VCB committed:
Interaction events v1


Interaction Events V1

More implementation

Implement server block place


Loicense


Javadoc for server block place


Structure for attack block events, raycasting for air events


Implement server interact block evt


Fix event name order, implement some more


server attack/interact events


Add item and air event classes


Remember some of the class renames


Add module lifecycles


deoverfinalization

L2750558108 commented:
Why dont the BlockPlaceEvent merge? I really need it. Or is there any ways to replace it?
